src/auth.rs

```diff
@@ -28,18 +35,48 @@ impl Authentication {
             Authentication::None => AuthType::NoAuth,
             Authentication::Direct(_) => AuthType::Direct,
             Authentication::UnixPeerCredentials => AuthType::UnixPeerCredentials,
+            #[cfg(feature = "spiffe-auth")]
+            Authentication::JwtSvid => AuthType::JwtSvid,
         }
     }
 }
 
-impl From<&Authentication> for RequestAuth {
-    fn from(data: &Authentication) -> Self {
+impl TryFrom<&Authentication> for RequestAuth {
+    type Error = Error;
+
+    fn try_from(data: &Authentication) -> Result<Self> {
         match data {
-            Authentication::None => RequestAuth::new(Vec::new()),
-            Authentication::Direct(name) => RequestAuth::new(name.bytes().collect()),
+            Authentication::None => Ok(RequestAuth::new(Vec::new())),
+            Authentication::Direct(name) => Ok(RequestAuth::new(name.bytes().collect())),
             Authentication::UnixPeerCredentials => {
                 let current_uid = users::get_current_uid();
-                RequestAuth::new(current_uid.to_le_bytes().to_vec())
+                Ok(RequestAuth::new(current_uid.to_le_bytes().to_vec()))
+            }
+            #[cfg(feature = "spiffe-auth")]
+            Authentication::JwtSvid => {
+                use crate::error::ClientErrorKind;
+                use log::error;
+                use spiffe::workload::jwt::JWTClient;
+                use std::env;
+
+                let client = JWTClient::new(
+                    &env::var("SPIFFE_ENDPOINT_SOCKET").map_err(|e| {
+                        error!(
+                            "Cannot read the SPIFFE_ENDPOINT_SOCKET environment variable ({}).",
+                            e
+                        );
+                        Error::Client(ClientErrorKind::NoAuthenticator)
+                    })?,
+                    None,
+                    None,
+                );
+                let audience = String::from("parsec");
+
+                let result = client.fetch(audience).map_err(|e| {
+                    error!("Error while fetching the JWT-SVID ({}).", e);
+                    Error::Client(ClientErrorKind::Spiffe(e))
+                })?;
+                Ok(RequestAuth::new(result.svid().as_bytes().into()))
```

Comment:
Do you need to zeroize the contents of `result`, given that they contain a token?

Reply:
I will implement `Zeroize` on drop on the `Jwt` structure to make sure it is zeroized all the time! Will do that directly in the `rust-spiffe` structure.

Reply:
Did it [here](https://github.com/hug-dev/rust-spiffe/blob/7a6fd03f81fb194feaf5fe6c19e07db8843c6b72/spiffe/src/svid/jwt.rs#L21)!

Reply:
Cheers 💯